crates/typst-library/src/text/shaping.rs

```diff
@@ -430,35 +430,51 @@ impl<'a> ShapedText<'a> {
 
     /// Push a hyphen to end of the text.
     pub fn push_hyphen(&mut self, vt: &Vt) {
-        families(self.styles).find_map(|family| {
```

Comment:
Writing it like this _should_ do the same thing right? It's a bit simpler.

We can then also skip coverage check because it happens implicitly in the loop.
```rust
let world = vt.world;
let book = world.book();
let mut chain = families(self.styles)
    .map(|family| book.select(family.as_str(), self.variant))
    .chain(std::iter::once_with(|| book.select_fallback(None, self.variant, "-")))
    .flatten();

chain.find_map(|id| {
    let font = world.font(id)?;
    let ttf = font.ttf();
    let glyph_id = ttf.glyph_index('-')?;
    let x_advance = font.to_em(ttf.glyph_hor_advance(glyph_id)?);
    let range = self
        .glyphs
        .last()
        .map(|g| g.range.end..g.range.end)
        .unwrap_or_default();
    self.width += x_advance.at(self.size);
    self.glyphs.to_mut().push(ShapedGlyph {
        font,
        glyph_id: glyph_id.0,
        x_advance,
        x_offset: Em::zero(),
        y_offset: Em::zero(),
        adjustability: Adjustability::default(),
        range,
        safe_to_break: true,
        c: '-',
        span: (Span::detached(), 0),
    });
    Some(())
});
```

Reply:
Indeed. I updated the revised version to use the chained iterator, which is a great idea. Thank you.